examples/basics/scene/points_selection.py

```diff
@@ -0,0 +1,279 @@
+
+        # Add a text instruction
+        self.text = scene.visuals.Text('', pos=(self.canvas.size[0]/4.0,  20),
+                                       color='w', parent=self.canvas.scene)
+
+        self.tr = self.scatter.node_transform(self.view)
+
+        # Add a 3D axis to keep us oriented
+        self.axis = scene.visuals.XYZAxis(parent=self.view.scene)
+
+        # Set up for lasso drawing
+        self.line_pos = []
+        self.line = scene.visuals.Line(color='white', method='gl',
+                                       parent=self.canvas.scene)
+
+        # select
+        self.select_flag = False
+        self.select_pool = {'1': 'lasso', '2': 'rectangle',
+                            '3': 'ellipse', '4': 'pick'}
+        self.select_id = '1'  # default as 1
+        self.select_origin = (0, 0)
+
+    def event_connect(self, flag):
+        cam_events = self.view.camera._viewbox.events
+        cam_mouse_event = self.view.camera.viewbox_mouse_event
+        if flag:
+            cam_events.mouse_move.disconnect(cam_mouse_event)
+            cam_events.mouse_press.disconnect(cam_mouse_event)
+            cam_events.mouse_release.disconnect(cam_mouse_event)
+            cam_events.mouse_wheel.disconnect(cam_mouse_event)
+        else:
+            cam_events.mouse_move.connect(cam_mouse_event)
+            cam_events.mouse_press.connect(cam_mouse_event)
+            cam_events.mouse_release.connect(cam_mouse_event)
+            cam_events.mouse_wheel.connect(cam_mouse_event)
+
+    def mark_selected(self):
+        # Change the color of the selected point
+
+        self.facecolor[self.facecolor[:, 1] != 1.0] = self.white
+        self.scatter.set_data(self.data, face_color=self.facecolor,
+                              size=self.ptsize)
+        for i in self.selected:
+            self.facecolor[i] = [1.0, 0.0, 0.0, 1]
+
+        self.scatter.set_data(self.data, face_color=self.facecolor,
+                              size=self.ptsize)
+        self.scatter.update()
+
+    def on_key_press(self, event):
+        # Set select_flag and instruction text
+
+        if event.text in self.select_pool.keys():
+            if not self.select_flag:
+                self.text.text = '%s select mode, press %s to switch' \
+                                 % (self.select_pool[event.text], event.text)
+                self.select_flag = True
+            else:
+                self.text.text = 'View mode, press %s to switch' \
+                                 % event.text
+                self.select_flag = False
+            self.event_connect(self.select_flag)
+            self.select_id = event.text
+
+    def on_mouse_press(self, event):
+        # Realize picking functionality and set origin mouse pos
+
+        if event.button == 1 and self.select_flag:
+            if self.select_id == '4':
+                # Ray intersection on the CPU to highlight selected point(s)
+
+                data = self.tr.map(self.data)[:, :2]
```

Comment:
I'm trying to adapt your example into my own code. `self.tr.map` seems to return homogeneous coordinates. You need to change this to:

```
data = self.tr.map(self.data)
data = data[:, :2] / data[:, 2:3]
```

From testing `data[:, 2]` and `data[:, 3]` are (approximately) the same, so these are 4d homogeneous coords for 2d points.